soda_machine.rb

```diff
@@ -7,12 +7,20 @@ def initialize(args = {})
   end
 
   def current_inventory_count
+    @sodas.size
   end
 
   def find_soda(soda_brand)
+    @sodas.find{|soda| soda.brand == soda_brand}
   end
 
   def sell(soda_brand)
+    this_soda = find_soda(soda_brand)
+    if this_soda != nil
+      @sodas.delete_at(@sodas.index(this_soda))
```

Comment:
Hi @amgando!

Thanks again for the timely code review. Below is my reasoning for using #delete_at instead of #delete - please do let me know if there are any flaws in my logic. :-)

According to the [ruby docs](http://ruby-doc.org/core-2.3.1/Array.html#method-i-delete), #delete would remove all instances of this_soda, and not just one. Our [spec file](https://github.com/sf-otters-2017/phase-1-checkpoint-challenge-2-soda-machine/blob/master/spec/soda_machine_spec.rb) initializes with two instances of pepsi, so I wanted to avoid writing code that would empty the machine of pepsi altogether if a user bought a pepsi. (Ironically, the spec file doesn't actually test for purchasing Pepsi, which is the _only_ soda that was initialized twice.)

I'm pretty confident that `@sodas.delete_at(@sodas.index(this_soda))` accounts for the issue of multiple sodas - but again, we never tested for it.